powerauth-restful-security-javaee/src/main/java/io/getlime/security/powerauth/rest/api/jaxrs/provider/PowerAuthEncryptionProvider.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * PowerAuth integration libraries for RESTful API applications, examples and
+ * related software components
+ *
+ * Copyright (C) 2018 Lime - HighTech Solutions s.r.o.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package io.getlime.security.powerauth.rest.api.jaxrs.provider;
+
+import io.getlime.security.powerauth.http.PowerAuthEncryptionHttpHeader;
+import io.getlime.security.powerauth.http.validator.InvalidPowerAuthHttpHeaderException;
+import io.getlime.security.powerauth.http.validator.PowerAuthEncryptionHttpHeaderValidator;
+import io.getlime.security.powerauth.rest.api.base.encryption.PowerAuthEciesEncryption;
+import io.getlime.security.powerauth.rest.api.base.exception.PowerAuthEncryptionException;
+import io.getlime.security.powerauth.rest.api.base.provider.PowerAuthEncryptionProviderBase;
+import io.getlime.security.powerauth.rest.api.jaxrs.encryption.PowerAuthEciesEncryptionImpl;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Implementation of PowerAuth encryption provider.
+ *
+ * @author Roman Strobl, roman.strobl@wultra.com
+ *
+ */
+public class PowerAuthEncryptionProvider extends PowerAuthEncryptionProviderBase  {
+
+    @Override
+    public PowerAuthEciesEncryption validateEciesEncryption(String encryptionHttpHeader) throws PowerAuthEncryptionException {
```

Comment:
As mentioned with the encryption interface couple lines above, this seems a bit redundant to me... We were making an abstraction for authentication so that we are able to accommodate different requirements for authentication handling by Spring and JAX-RS. For E2EE, this is not necessary.